<p>Our Dataset is composed of mental health and frequency of mental health disorders in the tech workplace.</p>
<h3>Question to responde: Taking a medical leave for a mental health condition?</h3>
<p>What the most influencing features for then to take a leave or not.</p>
<p>In this dataset the response for talking a leave is:</p>
<ul>
    <li>Don't know: 563</li>
    <li>Somewhat easy: 266</li>
    <li>Very easy: 206</li>
    <li>Somewhat difficult: 126</li>
</ul>
<p> The model is going to be train and an API in going to be available to test new cases.</p>
    


## Data set Mental Health in TI

<p>"This dataset is from a 2014 survey that measures attitudes towards mental health and frequency of mental health disorders in the tech workplace."(From Kanggle)</p>
<p> link: <a href="https://www.kaggle.com/datasets/osmi/mental-health-in-tech-survey?resource=download&select=survey.csv">Data source Kanggle</a></p>
<p>Version 1.2 - 28/05/2022</p>

In [23]:
## imports
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

import warnings
warnings.filterwarnings('ignore')

### 1 Data Collection

In [2]:
df = pd.read_csv("data/survey.csv")
df['leave'].value_counts()

Don't know            563
Somewhat easy         266
Very easy             206
Somewhat difficult    126
Very difficult         98
Name: leave, dtype: int64

### 2 Data Understanding

In [3]:
df.shape

(1259, 27)

In [4]:
df.head()

Timestamp  Age  Gender         Country state self_employed  \
0  2014-08-27 11:29:31   37  Female   United States    IL           NaN   
1  2014-08-27 11:29:37   44       M   United States    IN           NaN   
2  2014-08-27 11:29:44   32    Male          Canada   NaN           NaN   
3  2014-08-27 11:29:46   31    Male  United Kingdom   NaN           NaN   
4  2014-08-27 11:30:22   31    Male   United States    TX           NaN   

  family_history treatment work_interfere    no_employees  ...  \
0             No       Yes          Often            6-25  ...   
1             No        No         Rarely  More than 1000  ...   
2             No        No         Rarely            6-25  ...   
3            Yes       Yes          Often          26-100  ...   
4             No        No          Never         100-500  ...   

                leave mental_health_consequence phys_health_consequence  \
0       Somewhat easy                        No                      No   
1          Don't know                     Maybe                      No   
2  Somewhat difficult                        No                      No   
3  Somewhat difficult                       Yes                     Yes   
4          Don't know                        No                      No   

      coworkers supervisor mental_health_interview phys_health_interview  \
0  Some of them        Yes                      No                 Maybe   
1            No         No                      No                    No   
2           Yes        Yes                     Yes                   Yes   
3  Some of them         No                   Maybe                 Maybe   
4  Some of them        Yes                     Yes                   Yes   

  mental_vs_physical obs_consequence comments  
0                Yes              No      NaN  
1         Don't know              No      NaN  
2                 No              No      NaN  
3                 No             Yes      NaN  
4         Don't know              No      NaN  

[5 rows x 27 columns]

In [5]:
#Tipos das colunas
df.dtypes

Timestamp                    object
Age                           int64
Gender                       object
Country                      object
state                        object
self_employed                object
family_history               object
treatment                    object
work_interfere               object
no_employees                 object
remote_work                  object
tech_company                 object
benefits                     object
care_options                 object
wellness_program             object
seek_help                    object
anonymity                    object
leave                        object
mental_health_consequence    object
phys_health_consequence      object
coworkers                    object
supervisor                   object
mental_health_interview      object
phys_health_interview        object
mental_vs_physical           object
obs_consequence              object
comments                     object
dtype: object

In [6]:
# Quantidades de linhas do dataset
len(df)

1259

In [7]:
#Soma nulos por Coluna
df.isnull().sum()

Timestamp                       0
Age                             0
Gender                          0
Country                         0
state                         515
self_employed                  18
family_history                  0
treatment                       0
work_interfere                264
no_employees                    0
remote_work                     0
tech_company                    0
benefits                        0
care_options                    0
wellness_program                0
seek_help                       0
anonymity                       0
leave                           0
mental_health_consequence       0
phys_health_consequence         0
coworkers                       0
supervisor                      0
mental_health_interview         0
phys_health_interview           0
mental_vs_physical              0
obs_consequence                 0
comments                     1095
dtype: int64

In [8]:
# Soma todos os NaN do dataset
df.isna().sum().sum()

1892

### 3 Data Preparation

#### Dealing with missing values
<p>Missing values in columns</p>
<ul>
<li>state - 515</li>
<li>self_employed - 18</li>
<li>work_interfere - 264</li>
<li>comments - 1095</li>
</ul>



In [9]:
# Put "Not informed" in NaN to state column
# Column: state - 515 NaN
df['state'].replace(np.nan, "Not informed", inplace=True)

In [10]:
# Column: work_interfere - 264 NaN
df['work_interfere']

0           Often
1          Rarely
2          Rarely
3           Often
4           Never
          ...    
1254          NaN
1255        Often
1256    Sometimes
1257          NaN
1258    Sometimes
Name: work_interfere, Length: 1259, dtype: object

In [11]:
# Sum NaN Again
df.isnull().sum()

Timestamp                       0
Age                             0
Gender                          0
Country                         0
state                           0
self_employed                  18
family_history                  0
treatment                       0
work_interfere                264
no_employees                    0
remote_work                     0
tech_company                    0
benefits                        0
care_options                    0
wellness_program                0
seek_help                       0
anonymity                       0
leave                           0
mental_health_consequence       0
phys_health_consequence         0
coworkers                       0
supervisor                      0
mental_health_interview         0
phys_health_interview           0
mental_vs_physical              0
obs_consequence                 0
comments                     1095
dtype: int64

#### Dropping variables

In [12]:
# Drop column: comments
df.drop(["comments"], axis=1, inplace=True)

In [13]:
df.columns

Index(['Timestamp', 'Age', 'Gender', 'Country', 'state', 'self_employed',
       'family_history', 'treatment', 'work_interfere', 'no_employees',
       'remote_work', 'tech_company', 'benefits', 'care_options',
       'wellness_program', 'seek_help', 'anonymity', 'leave',
       'mental_health_consequence', 'phys_health_consequence', 'coworkers',
       'supervisor', 'mental_health_interview', 'phys_health_interview',
       'mental_vs_physical', 'obs_consequence'],
      dtype='object')

In [14]:
# Sum of null values
df.isnull().sum()

Timestamp                      0
Age                            0
Gender                         0
Country                        0
state                          0
self_employed                 18
family_history                 0
treatment                      0
work_interfere               264
no_employees                   0
remote_work                    0
tech_company                   0
benefits                       0
care_options                   0
wellness_program               0
seek_help                      0
anonymity                      0
leave                          0
mental_health_consequence      0
phys_health_consequence        0
coworkers                      0
supervisor                     0
mental_health_interview        0
phys_health_interview          0
mental_vs_physical             0
obs_consequence                0
dtype: int64

### Exploração Analítica (EDA)

In [24]:
categorical_features = df.columns[df.dtypes == object]
numerical_features = df.columns[df.dtypes != object]

In [25]:
categorical_features

Index(['Timestamp', 'Gender', 'Country', 'state', 'self_employed',
       'family_history', 'treatment', 'work_interfere', 'no_employees',
       'remote_work', 'tech_company', 'benefits', 'care_options',
       'wellness_program', 'seek_help', 'anonymity', 'leave',
       'mental_health_consequence', 'phys_health_consequence', 'coworkers',
       'supervisor', 'mental_health_interview', 'phys_health_interview',
       'mental_vs_physical', 'obs_consequence'],
      dtype='object')

In [26]:
numerical_features

Index(['Age'], dtype='object')

In [63]:
df['seek_help'].value_counts( normalize=True ) * 100

No            51.310564
Don't know    28.832407
Yes           19.857029
Name: seek_help, dtype: float64

#### Change Categorical values for Numeric

In [62]:
df['self_employed'] = df['self_employed'].map({'Yes':1, 'No':0})
df['family_history'] = df['family_history'].map({'Yes':1,'No':0})
df['treatment'] = df['treatment'].map({'Yes':1,'No':0})
df['remote_work'] = df['remote_work'].map({'Yes':1,'No':0})
df['tech_company'] = df['tech_company'].map({'Yes':1,'No':0})
df['obs_consequence'] = df['obs_consequence'].map({'Yes':1,'No':0})

#### More than two options in Categorical values - use pd.factorize

In [65]:
df['seek_help_num'] = pd.factorize(df['seek_help'])[0]

In [66]:
df['seek_help_num'].value_counts()

2    646
1    363
0    250
Name: seek_help_num, dtype: int64

In [75]:
df[['seek_help_num','seek_help']]

seek_help_num   seek_help
0                 0         Yes
1                 1  Don't know
2                 2          No
3                 2          No
4                 1  Don't know
...             ...         ...
1254              2          No
1255              2          No
1256              2          No
1257              2          No
1258              2          No

[1259 rows x 2 columns]